lib/vtls/vtls.c

```diff
@@ -338,7 +340,9 @@ Curl_ssl_connect_nonblocking(struct Curl_easy *data, struct connectdata *conn,
   /* mark this is being ssl requested from here on. */
   conn->ssl[sockindex].use = TRUE;
   result = Curl_ssl->connect_nonblocking(data, conn, sockindex, done);
-  if(!result && *done)
+  if(result)
+    conn->ssl[sockindex].use = FALSE;
```

Comment:
Wouldn't it make even more sense to instead set it TRUE after a successful return?

Reply:
Probably, but I did not check if it is needed to be true while negotiating.

Reply:
Right, probably better to place it safe.